### DISCUSSION TITLE
Using implicit integrator with Unity plugin

### DISCUSSION THREAD

**Discussion by Balint-H** [google-deepmind/mujoco#368] [Asking for Help] [answered]
I've been running into instability with velocity actuation so I'm looking forward to using the implicit-in-velocity integrator. I've heard that eventually it will come to the Unity plugin as well, but excited to try it out, I poked around in the plugin's scripts and downloaded the latest release. Is adding this third option to the integrator types in the MjGlobalSettings sufficient to build the scene with the appropriate integrator?:

```
public enum IntegratorType {
  Euler,
  RK4,
  @implicit
}
```

@ symbol used so `implicit` is interpreted verbatim, since implicit is a keyword in C#. The scene runs with this selected, but not with any other, non-existent integrator names, which suggests to me that its working right.

On another note, I wanted to ask about the relation between this integrator and "Stable PD" controllers (as described by Tan, Liu and Turk). With this integrator selected, are PD controllers automatically SPD controllers? Or is this integrator more computationally costly than just using SPD with Euler?

Thanks!

**Comment by erez-tom** [comment, marked as answer]
Simulation stability in general could be improved by the implicit integrator; but for instabilities due to actuation, you might be able to benefit from integrated-velocity actuators, see [here](https://github.com/deepmind/mujoco/commit/6048a55dafe59d01ce2e497cfde7ff018212dab8).
BTW, I just upgraded the C# bindings to the latest version (so things should work better, but let me know if you hit any issues).

**Reply by Balint-H** [reply]
Are intvelocity actuators available with the Unity plugin in the most recent commit on `main` as well? I don't see them as an option in the Inspector, nor in the component source.
